sway-fmt-v2/src/config/manifest.rs

```diff
@@ -0,0 +1,170 @@
+use anyhow::{anyhow, Result};
+use forc_util::{find_parent_dir_with_file, println_yellow_err};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+pub use crate::error::FormatterError;
+use crate::{
+    config::{
+        comments::Comments, expr::Expressions, heuristics::Heuristics, imports::Imports,
+        items::Items, lists::Lists, literals::Literals, ordering::Ordering, user_def::Structures,
+        user_opts::*, whitespace::Whitespace,
+    },
+    constants::SWAY_FORMAT_FILE_NAME,
+};
+
+/// A finalized `swayfmt` config.
+#[derive(Debug)]
+pub struct Config {
+    pub whitespace: Whitespace,
+    pub imports: Imports,
+    pub ordering: Ordering,
+    pub items: Items,
+    pub lists: Lists,
+    pub literals: Literals,
+    pub expressions: Expressions,
+    pub heuristics: Heuristics,
+    pub structures: Structures,
+    pub comments: Comments,
+}
+
+/// A direct mapping to an optional `swayfmt.toml`.
+#[derive(Serialize, Deserialize, Debug, Copy, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct ConfigOptions {
+    pub whitespace: Option<WhitespaceOptions>,
+    pub imports: Option<ImportsOptions>,
+    pub ordering: Option<OrderingOptions>,
+    pub items: Option<ItemsOptions>,
+    pub lists: Option<ListsOptions>,
+    pub literals: Option<LiteralsOptions>,
+    pub expressions: Option<ExpressionsOptions>,
+    pub heuristics: Option<HeuristicsOptions>,
+    pub structures: Option<StructuresOptions>,
+    pub comments: Option<CommentsOptions>,
+}
+
+impl Config {
+    /// The default setting of `swayfmt`'s `Config`.
+    ///
+    pub fn default() -> Self {
+        Self {
+            whitespace: Whitespace::default(),
+            imports: Imports::default(),
+            ordering: Ordering::default(),
+            items: Items::default(),
+            lists: Lists::default(),
+            literals: Literals::default(),
+            expressions: Expressions::default(),
+            heuristics: Heuristics::default(),
+            structures: Structures::default(),
+            comments: Comments::default(),
+        }
+    }
```

Comment:
These `default() -> Self` constructors should be done with the `Default` trait instead, i.e.

```rust
impl Default for Config {
    fn default() -> Self {
        // ...
    }
}
```

However, seeing as you have already implemented `Default` for each of the sub-config types (i.e. `Whitespace`, `Imports`, etc), then you can use Rust's short-hand derive to generate the implementation instead, e.g.

```rust
#[derive(Debug, Default)]
pub struct Config {
```